src/provenance_gc.rs

```diff
@@ -162,6 +162,17 @@ impl VisitProvenance for crate::MiriInterpCx<'_> {
     }
 }
 
+impl<A, B> VisitProvenance for (A, B)
+where
```

Comment:
Please move this up to `Option<T>` so that we keep the impls for non-Miri types together.